chain/btcd.go

```diff
@@ -593,6 +593,38 @@ func (c *RPCClient) MapRPCErr(rpcErr error) error {
 		}
 	}
 
+	// If no matching error is found, we try to match it to an older
+	// version of `btcd`.
+	//
+	// Get the backend's version.
+	backend, bErr := c.BackendVersion()
+	if bErr != nil {
+		// If there's an error getting the backend version, we return
+		// the original error and the backend error.
+		return fmt.Errorf("%w: %v, failed to get backend version %v",
+			ErrUndefined, rpcErr, bErr)
+	}
+
+	// If this version supports `testmempoolaccept`, it must be v0.24.2 and
+	// above. In this case, we will return the original error.
+	//
+	// NOTE: `testmempoolaccept` is implemented in v0.24.1, but this
+	// version was never tagged, which means it must be v0.24.2 when it's
+	// supported.
+	if backend.SupportTestMempoolAccept() {
+		// Not matched, return the original error wrapped.
+		return fmt.Errorf("%w: %v", ErrUndefined, rpcErr)
+	}
+
+	// If the backend is older than v0.24.2, we will try to match the error
+	// to the older version of `btcd`.
+	for btcdErr, matchedErr := range BtcdErrMapPre2402 {
```

Comment:
nit: it could be slightly cleaner if we move this loop inside `if !backend.SupportTestMempoolAccept()` so that if the version is >= v0.24.2 we just return the error at the end.

Reply:
yeah was thinking about not nesting `if`s, but I guess it's cleaner so now changed.